pkg/userdata/centos/userdata.go

```diff
@@ -183,71 +169,70 @@ write_files:
     #     mls - Multi Level Security protection.
     SELINUXTYPE=targeted
 
-- path: "/etc/sysconfig/kubelet-overwrite"
-  content: |
-    KUBELET_DNS_ARGS=
-    KUBELET_EXTRA_ARGS=--authentication-token-webhook=true \
-      {{- if .CloudProvider }}
-      --cloud-provider={{ .CloudProvider }} \
-      --cloud-config=/etc/kubernetes/cloud-config \
-      {{- end}}
-      --hostname-override={{ .MachineSpec.Name }} \
-      --read-only-port=0 \
-      --protect-kernel-defaults=true \
-      --cluster-dns={{ ipSliceToCommaSeparatedString .ClusterDNSIPs }} \
-      --cluster-domain=cluster.local
-
-{{- if semverCompare "<1.11.0" .KubeletVersion }}
-- path: "/etc/systemd/system/kubelet.service.d/20-extra.conf"
-  content: |
-    [Service]
-    EnvironmentFile=/etc/sysconfig/kubelet
-{{- end }}
-
-- path: "/etc/kubernetes/cloud-config"
-  content: |
-{{ if ne .CloudConfig "" }}{{ .CloudConfig | indent 4 }}{{ end }}
-
-- path: "/usr/local/bin/setup"
+- path: "/opt/bin/setup"
   permissions: "0777"
   content: |
     #!/bin/bash
     set -xeuo pipefail
+
     setenforce 0 || true
+
+    # As we added some modules and don't want to reboot, restart the service 
+    systemctl restart systemd-modules-load.service
     sysctl --system
 
     yum install -y docker-1.13.1 \
-      kubelet-{{ .KubeletVersion }} \
-      kubeadm-{{ .KubeletVersion }} \
       ebtables \
       ethtool \
       nfs-utils \
       bash-completion \
-      sudo
+      sudo \
+      socat \
+      wget
 
-    cp /etc/sysconfig/kubelet-overwrite /etc/sysconfig/kubelet
+    # Download all required binaries
+    /opt/bin/download_binaries
```

Comment:
Can't we simply remove the shebang from the `downloadBinariesScript` template and embed it here with a `indent 4`?

Reply:
Inlined download script with https://github.com/kubermatic/machine-controller/pull/351/commits/0221895fac6284bffcd9088ae2a3ebe379e0795e